helix-vcs/src/git.rs

```diff
@@ -61,6 +76,36 @@ impl Git {
 
         Ok(res)
     }
+
+    /// Emulates the result of running `git status` from the command line.
+    fn status(repo: &Repository) -> Result<Box<dyn Iterator<Item = Result<FileChange>>>> {
```

Comment:
For abstracting over iteration like this I think external iteration instead of internal iteration is easier.

Instead of returning a dynamic dispatch iter here, you should pass a closure to this function. Than this function spawns the thread (with tokio), drives the iterator and calls the provided closure.

This also removes the need to implememt iterator which is always slightly messy/slow.

Reply:
You're right. I was originally planning to mimic the optimistic sync trial pattern from `file_picker`, which I believe necessitate an iterator. Now that we don't do it we can use a closure instead. Will fix. Thanks!